tfjs-core/src/ops/draw_test.ts

```diff
@@ -16,89 +16,201 @@
  */
 
 import * as tf from '../index';
-import {BROWSER_ENVS, describeWithFlags} from '../jasmine_util';
+import {Constraints, describeWithFlags} from '../jasmine_util';
 import {expectArraysClose, expectArraysEqual} from '../test_util';
 
-class MockContext {
-  data: ImageData;
-
-  getImageData() {
-    return this.data;
-  }
-
-  putImageData(data: ImageData, x: number, y: number) {
-    this.data = data;
+function readPixelsFromCanvas(
+    canvas: OffscreenCanvas, contextType: string, width: number,
+    height: number) {
+  let actualData;
+  if (contextType === '2d') {
+    const ctx = canvas.getContext(contextType);
+    actualData = ctx.getImageData(0, 0, height, width).data;
+  } else {
+    const offscreenCanvas = new OffscreenCanvas(height, width);
+    const ctx = offscreenCanvas.getContext('2d');
+    ctx.drawImage(canvas, 0, 0);
+    actualData = new Uint8ClampedArray(
+        ctx.getImageData(0, 0, ctx.canvas.width, ctx.canvas.height).data);
   }
+  return actualData;
 }
 
-class MockCanvas {
-  context: MockContext;
+const BROWSER_CPU_WEBGPU_ENVS: Constraints = {
+  predicate: (env) =>
```

Comment:
IIUC, the Core is not supposed to know the backend, so we could filter out tests in the setup_tests of backends. In this case, we just need to filter out WebGL (done) and WASM backends. You could add an 'exclude' in tfjs-backend-wasm/src/setup_test.ts.

Reply:
It seems wasm uses a whitelist: https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-wasm/src/setup_test.ts#L42, so we do not need to exclude it at all?